Support indentless nested array

abicky opened the pull request:
This PR resolves https://github.com/google/yamlfmt/issues/167.
I've confirmed the behavior using the following code:

```go
package main

import (
	"bytes"
	"fmt"

	"github.com/braydonk/yaml"
)

var data = `
a:
- - 1
  - 2`

func main() {
	reader := bytes.NewReader([]byte(data))
	decoder := yaml.NewDecoder(reader)
	var node yaml.Node
	err := decoder.Decode(&node)
	if err != nil {
		panic(err)
	}

	var buf bytes.Buffer
	encoder := yaml.NewEncoder(&buf)
	encoder.SetIndent(2)
	encoder.SetIndentlessBlockSequence(true)

	if err := encoder.Encode(&node); err != nil {
		panic(err)
	}
	fmt.Print(string(buf.Bytes()))
}
```

## Before

```
a:
-
- 1
- 2
```

## After

```
a:
- - 1
  - 2
```

abicky committed:
Support indentless nested array

This commit resolves https://github.com/google/yamlfmt/issues/167.

braydonk commented:
Thanks for the PR @abicky, this slipped past me when it came in over the holidays so I forgot to get it reviewed. I will try to review it within the next couple days.

braydonk reviewed (approved):
This seems to work, thank you for the fix!

If possible, could you quickly add a comment to `emitterc.go` explaining why the fix works for future reference?

abicky commented:
Thank you for reviewing it!

> If possible, could you quickly add a comment to emitterc.go explaining why the fix works for future reference?

Sure! I thought about the comment and the following is what I came up with. Is this what you expected? If so, I will add the commit.

```diff
diff --git a/emitterc.go b/emitterc.go
index 660368b..d67926a 100644
--- a/emitterc.go
+++ b/emitterc.go
@@ -243,6 +243,9 @@ func yaml_emitter_increase_indent(emitter *yaml_emitter_t, flow, indentless bool
                }
        } else {
                if emitter.states[len(emitter.states)-1] == yaml_EMIT_BLOCK_SEQUENCE_ITEM_STATE {
+                       // Currently, we assume that indentless can be true for only block sequences
+                       // and this case is the first item of a nested indentless block sequence,
+                       // which means the item must be indented more.
                        emitter.indent += 2
                }
        }
```

braydonk commented:
Terribly sorry @abicky, I was running an update script on my machine and accidentally force pushed `v3` with an old commit which screwed this PR up. :face_exhaling: 

I restored the branch, but this PR needs a rebase to fix the history.

In your fork, here's what you'll need to run:
1. `git checkout v3`
2. `git remote add upstream git@github.com:braydonk/yaml.git`
3. `git fetch upstream`
4. `git rebase upstream/v3`
5. `git checkout support-indentless-nested-array`
6. `git rebase -i v3`
7. Drop all commits except for c989aa2
8. `git push --force-with-lease`
This should put your local copy in a normal state as well as fixing the PR here.

Sorry about this, I didn't know my fork didn't realize I hadn't properly set branch protections here and my script I ran was way too destructive.

braydonk commented:
The comment in https://github.com/braydonk/yaml/pull/18#issuecomment-2594548965 looks good to me, if you could add it that would be great!

abicky committed:
Add comment to yaml_emitter_increase_indent for indentless nested array

abicky commented:
No problem! I've rebased the branch onto the v3 branch and added the new commit https://github.com/braydonk/yaml/pull/18/commits/b45d4d3794e053727468577cf4952bb59bd19127 that includes the comment.

braydonk reviewed (approved):
Thanks for your patience, and thanks for the fix!

I will get this released and brought into yamlfmt as soon as I'm able.
